src/software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic_test.py

```diff
@@ -0,0 +1,364 @@
+                    tbots_cpp.Field.createSSLDivisionBField().enemyHalf(),
+                    tbots_cpp.Circle(ball_initial_pos, 1),
+                ]
+            ),
+            NeverExcessivelyDribbles(),
+        ]
+    ]
+
+    # Eventually Validation
+    eventually_validation_sequence_set = [
+        [
+            # Crease defender should be near the defense area
+            RobotEventuallyEntersRegion(
+                regions=[
+                    tbots_cpp.Field.createSSLDivisionBField()
+                    .friendlyDefenseArea()
+                    .expand(0.25)
+                ]
+            ),
+        ]
+    ]
+
+    simulated_test_runner.run_test(
+        setup=setup,
+        inv_eventually_validation_sequence_set=eventually_validation_sequence_set,
+        inv_always_validation_sequence_set=always_validation_sequence_set,
+        ag_eventually_validation_sequence_set=eventually_validation_sequence_set,
+        ag_always_validation_sequence_set=always_validation_sequence_set,
+        test_timeout_s=4,
+    )
+
+
+friendlyDefenseAreaFrontCenter = tbots_cpp.Point(
+    tbots_cpp.Field.createSSLDivisionBField()
+    .friendlyDefenseArea()
+    .posXPosYCorner()
+    .x(),
+    tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea().centre().y(),
+)
+
+
+@pytest.mark.parametrize(
+    "blue_bots, yellow_bots, ball_initial_pos, ball_initial_velocity, should_chip",
+    [
+        # Test auto chip over enemy
+        (
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.5, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(1.1, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.9, 0),
+            tbots_cpp.Vector(-2, 0),
+            True,
+        ),
+        # Test block, auto chip off, enemy far
+        (
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.5, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(5, 0),
+            tbots_cpp.Point(0, 0),
+            tbots_cpp.Vector(-6, 0),
+            False,
+        ),
+        # Test block, auto chip off, enemy close, facing net
+        (
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(1.5, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.2, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.9, 0),
+            tbots_cpp.Vector(2, 0),
+            False,
+        ),
+        # Test block, auto chip off, enemy far, facing net
+        (
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(3, 0),
+            tbots_cpp.Point(3, 0),
+            friendlyDefenseAreaFrontCenter + tbots_cpp.Vector(0.9, 0),
+            tbots_cpp.Vector(2, 0),
+            False,
+        ),
+    ],
+)
+def test_crease_autochip(
+    blue_bots,
+    yellow_bots,
+    ball_initial_pos,
+    ball_initial_velocity,
+    should_chip,
+    simulated_test_runner,
+):
```

Comment:
One of these tests failed for me locally: `test_crease_autochip[blue_bots0-yellow_bots0-ball_initial_pos0-ball_initial_velocity0-True]` because `Failed: EventuallyTrueValidation: Check if the ball is chipped failed`. I think 0.01 is a very small threshold for chips. we could make it atleast 0.05